identity/registry/registry_contract.go

```diff
@@ -40,19 +40,23 @@ type registryStorage interface {
 	GetAll() ([]StoredRegistrationStatus, error)
 }
 
+type getLocation func() string
+
 type contractRegistry struct {
 	hermesAddress   common.Address
 	storage         registryStorage
-	stop            chan struct{}
-	once            sync.Once
 	publisher       eventbus.Publisher
-	lock            sync.Mutex
 	ethC            *paymentClient.ReconnectableEthClient
 	registryAddress common.Address
+	getLocation     getLocation
+
+	stop chan struct{}
+	lock sync.Mutex
+	once sync.Once
 }
 
 // NewIdentityRegistryContract creates identity registry service which uses blockchain for information
-func NewIdentityRegistryContract(ethClient *paymentClient.ReconnectableEthClient, registryAddress, hermesAddress common.Address, registryStorage registryStorage, publisher eventbus.Publisher) (*contractRegistry, error) {
+func NewIdentityRegistryContract(ethClient *paymentClient.ReconnectableEthClient, registryAddress, hermesAddress common.Address, registryStorage registryStorage, publisher eventbus.Publisher, gl getLocation) (*contractRegistry, error) {
```

Comment:
Dont couple this component to location directly, but rather depend on location event with `Subscribe(eb eventbus.Subscriber) error`

Reply:
Country is needed for registration metric, not necessarily for all registration subscribers.
Consider appending it only in metric sending.

Reply:
We've talked about this. What I think we should do is remake how location detection works allowing our metrics service to get it any time it wants through the event bus and then remove such dependencies.

What sucks now, is "prefilling" the bus with a location object is kinda bad and hacky.

Reply:
I made node because of 2 concerns:
1. Original node location does not change, so doing request for each registration is excess
2. We have plans location changes event based, so this will be +1 place to refactor

Reply:
Second point is true, yes this is +1 place to refactor. But I still think that depending on other parts of the code to call detect location just so that we know the location before registering identity is sort of strange and might fail but so be it, I'll make the changes 🤷